constantine/build.gradle

```
@@ -0,0 +1,171 @@
+plugins {
+    id 'java-library'
+    id 'maven-publish'
+    id 'com.jfrog.artifactory' version '5.2.3'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'net.java.dev.jna:jna:5.12.1'
+    testImplementation 'com.google.guava:guava:31.1-jre'
+    testImplementation 'io.tmio:tuweni-bytes:2.4.2'
+    testImplementation 'junit:junit:4.13.2'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+task makeConstantineLib(type: Exec) {
+    description = 'Compiles the Constantine library'
+    workingDir = file('./constantine')
+    environment 'PATH', "${System.env.PATH}:${System.getenv('HOME')}/.nimble/bin"
+    environment 'CTT_LTO', 'false'
+    commandLine "${System.getenv('HOME')}/.nimble/bin/nimble", 'make_lib'
```

```suggestion
    commandLine "nimble", 'make_lib'
    inputs.dir file('./constantine/constantine')
    outputs.dir file('./constantine/lib')```
    
    gradle needs to know about inputs and outputs to reason about if/what to cache.  What is odd is I didn't have the same problem locally.  Removing the explicit path to nim (since you already added it to PATH) makes this build cleanly for me on my local